cmd/zed/zed.d/all-syslog.sh

```diff
@@ -5,6 +5,18 @@
 [ -f "${ZED_ZEDLET_DIR}/zed.rc" ] && . "${ZED_ZEDLET_DIR}/zed.rc"
 . "${ZED_ZEDLET_DIR}/zed-functions.sh"
 
+if [ -n "${ZED_SYSLOG_SUBCLASS_INCLUDE}" ]; then
+    case "${ZEVENT_SUBCLASS}" in
```

Comment:
I was unsure about this as well. In the end, all the use cases I came up with didn't require the full power of a regex engine--we're dealing with a well-known, fixed list of subclass names, after all. So I opted for an implementation without external dependencies, and only POSIX sh features. But if you prefer the grep variant, I'll happily revise the patch.

Reply:
@sckobras using only POSIX sh features doesn't seem to work properly though, https://github.com/zfsonlinux/zfs/pull/7007#issuecomment-355007909.

Reply:
@loli10K Yes, sorry. That one's already fixed locally with an extra `eval`. I just want to make sure if this is the way to go, or `grep` (or `#!/bin/bash`) would be the preferred approach.

Reply:
As long as you can address the issue @loli10K identified (and it sounds like you did) I'm OK with using the shell patterns.  I agree it would be nice not to have to rely on grep here is we can avoid it.